src/Passage.sol

```diff
@@ -72,15 +93,23 @@ contract HostPassage {
     ///      the Builder may transfer the cumulative tokenOut to the user in a single ExitFilled event.
     ///      The rollup STF will apply the user's exit transactions on the rollup up to the point that sum(tokenOut) is lte the ExitFilled amount.
     /// TODO: add option to fulfill ExitOrders with native ETH? or is it sufficient to only allow users to exit via WETH?
-    function fulfillExits(ExitOrder[] calldata orders) external {
+    function fulfillExits(ExitOrder[] calldata orders) external payable {
```

Comment:
why do we perform any checks at all here? given the purpose is for builder to transfer funds, we don't really need to safety check them. they are capable of doing that for themselves.

```
struct Fullfillment {
    // generated somewhere in rollup consensus, must commit to chain ID
    uint256 orderId; 

    address token;
    address recipient;
    uint256 amount;
}
```

Reply:
basically i think at least the Deadline check is unnecessary, as that part will be performed in the RU consensus

Reply:
> why do we perform any checks at all here? given the purpose is for builder to transfer funds, we don't really need to safety check them. they are capable of doing that for themselves.

i don't understand what you mean 👀 the thinking behind this fn is to emit events that the RU node can use to easily see that funds have been transferred in accordance with the intent expressed on L2. we perform checks to ensure that said events actually correspond to funds transfers

Reply:
Zenith doesnt need to perform any check that there is a corresponding RU tx that the builder is trying to fill

Reply:
(and indeed can't effectively perform any check like that)

Reply:
the builder is capable of making that check outside this system, if they want to. we dont need to make it mandatory for them by putting it in-system

Reply:
> ah do you just mean the Builder adds a block deadline

yes they can define a confirmby :)

Reply:
i removed the deadline check: https://github.com/init4tech/zenith/pull/11/commits/53982a5b09645a035536e889585ab67e04797032

Reply:
not sure about `orderId` - how would you propose handling that in the RU node?